app/resources/api/v2/state_change_resource.rb

```diff
@@ -15,14 +15,87 @@ module V2
     # or look at the [JSONAPI::Resources](http://jsonapi-resources.com/) package for Sequencescape's implementation
     # of the JSON:API standard.
     class StateChangeResource < BaseResource
-      immutable
+      ###
+      # Attributes
+      ###
 
-      attribute :previous_state
+      # @!attribute [rw] contents
+      #   @return [Array] Array of "contents" to fail, deciphered by the target.
+      attribute :contents
+
+      # @!attribute [w] customer_accepts_responsibility
+      #   @param value [Boolean] Sets whether the customer proceeded against advice and will still be charged in the
+      #     event of a failure.
+      #   @return [Void]
+      attribute :customer_accepts_responsibility
+
+      # @!attribute [r] previous_state
+      #   @return [String] The previous state of the target before this state change.
+      attribute :previous_state, readonly: true
+
+      # @!attribute [rw] reason
+      #   @return [String] The previous state of the target before this state change.
+      attribute :reason
+
+      # @!attribute [rw] target_state
+      #   @return [String] The state of the target after this state change.
+      #   @note This attribute is required.
       attribute :target_state
-      attribute :created_at
-      attribute :updated_at
 
-      has_one :labware
+      # @!attribute [w] target_uuid
+      #   This is declared for convenience where the target is not available to set as a relationship.
+      #   Setting this attribute alongside the `target` relationship will prefer the relationship value.
+      #   @param value [String] The UUID of the target labware this state change applies to.
+      #   @return [Void]
+      #   @see #target
+      attribute :target_uuid
+
+      def target_uuid=(value)
```

Comment:
Does this belong in here or in the model?

Reply:
This is how we've created other UUID based setters on the API.  Resources let you override the getter and/or setter for attributes, and this one is here to set the actual target on the model.  If we put this in the model, it would be to support the API and I don't think that's the right thing to do.

To be honest, users of the API should really be using the relationships to assign other resources as related, but Limber is very heavily architected around passing the UUID to the v1 API and it would create another hurdle to overcome if we don't have v2 API support the same approach.